extensions/git/src/autofetch.ts

```diff
@@ -19,7 +19,7 @@ function isRemoteOperation(operation: Operation): boolean {
 
 export class AutoFetcher {
 
-	private static readonly Period = 3 * 60 * 1000 /* three minutes */;
+	private static readonly Period = workspace.getConfiguration('git').get<number>('autoFetchPeriod') as number;
```

Comment:
A setting can change while Code is running. You need to listen to the event change.

Reply:
@joaomoreno , fixed the other two changes and working on this now.
I will close this PR and send a new one when it is completed :) 
Thanks.